dsc/src/resource_command.rs

```diff
@@ -14,14 +15,25 @@ use dsc_lib::{
 };
 use std::process::exit;
 
-pub fn get(dsc: &mut DscManager, resource: &str, input: &Option<String>, stdin: &Option<String>, format: &Option<OutputFormat>) {
+/// Get operation.
+///
+/// # Panics
+///
+/// Will panic if provider-based resource is not found.
+///
+pub fn get(dsc: &DscManager, resource_str: &str, input: &Option<String>, stdin: &Option<String>, format: &Option<OutputFormat>) {
     // TODO: support streaming stdin which includes resource and input
     let mut input = get_input(input, stdin);
-    let mut resource = get_resource(dsc, resource);
+
+    let Some(mut resource) = get_resource(dsc, resource_str) else {
+        error!("{}", DscError::ResourceNotFound(resource_str.to_string()).to_string());
+        return
+    };
+
     debug!("resource.type_name - {} implemented_as - {:?}", resource.type_name, resource.implemented_as);
-    if let Some(requires) = resource.requires {
-        input = add_type_name_to_json(input, resource.type_name);
-        resource = get_resource(dsc, &requires);
+    if let Some(requires) = &resource.requires {
+        input = add_type_name_to_json(input, resource.type_name.clone());
+        resource = get_resource(dsc, requires).unwrap();
```

Comment:
this is the line that can panic, you should do something like:

```rust
if let Some(resource) = get_resource(dsc, requires) else {
  error...
}
```

Reply:
Updated in other way. "let Some" syntax creates unused variable while I need this one to be overwritten here and used later.